synapse/handlers/room_batch.py

```diff
@@ -240,6 +241,7 @@ async def persist_state_events_at_start(
                     ),
                     event_dict,
                     outlier=True,
+                    historical=True,
```

Comment:
All of these changes are just piping `historical` down to `create_event`

```
create_and_send_nonmember_event
create_event
```

Reply:
whatever else happens, please please please make sure that any parameters are added to the docstrings *at each level*, with the intended purpose clearly described. 

Relatedly: as a general rule, I would say it is preferable for method parameters to describe a change in behaviour ("`inhibit_sync_to_clent`") rather than describe some bit of state that the function is free to interpret as it wishes: it's much clearer for a later reader to understand how things tie together, particularly when things then get modified later on. Obviously this isn't a hard-and-fast rule, but it's worth considering.

[Currently we have a bit of a mess in some places (`FederationEventHander.backfilled` being a prime example) where you have to dig through 150 methods to find out what a flag actually does, and then somehow reverse-engineer what it is *supposed* to do. Don't do that.]

Reply:
> please please please make sure that any parameters are added to the docstrings at each level, with the intended purpose clearly described.

Will do 🙇 

---

> Relatedly: as a general rule, I would say it is preferable for method parameters to describe a change in behaviour ("inhibit_sync_to_clent") rather than describe some bit of state that the function is free to interpret as it wishes

Seems reasonable. I've created a separate issue to track this https://github.com/matrix-org/synapse/issues/11300